### PR TITLE
chore(core): Moves settings.php and views.php to prevent MultiViews issues

### LINKED ISSUE #8806: MultiViews with root settings.php breaks settings URLs
It is not possible to access www.site.com/settings/* if ```settings.php``` is located at the Elgg root.

This also could be a problem with other magic/common Elgg directories like "views" or "actions".

Possible solutions:

 1. Prefix with `elgg-` (`elgg-settings.php`, `elgg-views/`, etc...)
 1. Move `index.php` handler into subdir (e.g., `public/index.php` like Laravel). Means you have to do some more work to make sure that "/public" isn't part of your website's path. :cry: 
 1. Place into `elgg-config/` directory: (`elgg-config/settings.php`, `elgg-config/views/`, etc.).
 1. Rename settings to something better like `.env.php` (might be a good idea anyways, but we'd still need to address the `views/`, `actions/`, etc. conflict.)
 1. Make `engine/start.php` the default location for non-composer installs.
 1. Just `elgg-` prefix the settings file.

### PR DESCRIPTION
**Pull request by mrclay**
The default settings.php location is now inside elgg-config/ instead of the installation root. At the root, settings.php would interfere with settings URLs on Apache installations with MultiViews, and site owners can't always disable this.

views.php is moved inside engine/.

Fixes #8806

### CODE REVIEW

File: engine/lib/views.php [hunk 1]
```diff
@@ -1574,7 +1574,7 @@ function elgg_views_boot() {
 
 	// Declared views. Unlike plugins, Elgg's root views/ is never scanned, so Elgg cannot override
 	// these view traditional view files.
-	$file = dirname(dirname(__DIR__)) . '/views.php';
+	$file = dirname(__DIR__) . '/views.php';
```
Comment (juho-jaakkola) [inline engine/lib/views.php]: Is this correct?

Comment (juho-jaakkola) [inline engine/lib/views.php]: Ah, this PR also moves ```views.php``` to ```engine/views.php```.

**Comment by juho-jaakkola** [comment]
This approach is fine, I guess.

Let's mention somewhere why the dir has been named like that, so future devs don't have to wonder about it.

I have the feeling this RP should be tested quite extensively because it changes the path in so many places.

**Comment by mrclay** [comment]
> Let's mention somewhere why the dir has been named like that

Any idea where that should go?

**Comment by mrclay** [comment]
I think devs can use git tools like blame/history to find this commit.

**Comment by juho-jaakkola** [comment]
True. I'm so used to wondering about code added before Github, that I somethimes forget the Blame possibility. :)

**Comment by juho-jaakkola** [comment]
How confident are you that this PR won't break anything? I'm afraid I won't have time to do proper testing.

**Comment by mrclay** [comment]
Not super confident. I guess it needs manual installs in both setups we support and clicking around through every feature to be sure.

The idea of offering Elgg as a composer dependency is great. In practice it almost doubles our already large manual testing burden, has led to confusion for even longtime Elgg developers, and both at a time when core developer involvement is critically low.

**Comment by juho-jaakkola** [comment]
Installation as composer dependency is failing because the ```elgg-config/``` directory doesn't exist in the starter-project root.

**Comment by juho-jaakkola** [comment]
Doesn't that mean installer is also looking for ```settings.example.php``` from the ```elgg-config/``` dir in starter root?

**Comment by mrclay** [comment]
I've updated the installer to check the local elgg-config for write permissions, and also the message it gives you tells you the correct source location of the elgg.settings.php file (e.g. `"vendor/elgg/elgg/elgg-config/settings.example.php"` or `"elgg-config/settings.example.php"`).

After merging we'll need to add the elgg-config dir to starter-project but that should be it.

**Pull request merged**